pkgs/development/libraries/mesa/default.nix

```diff
@@ -7,7 +7,8 @@
 , libglvnd, libunwind
 , vulkan-loader, glslang
 , galliumDrivers ? ["auto"]
-, vulkanDrivers ? ["auto"]
+# upstream Mesa defaults to only enabling swrast (aka lavapipe) on aarch64 for some reason, so force building the others
+, vulkanDrivers ? if (stdenv.isLinux && stdenv.isAarch64) then [ "swrast" "broadcom" "freedreno" "panfrost" ] else [ "auto" ]
```

Comment:
It will be easier to maintain if you use `[ "auto" "broadcom" "freedreno" "panfrost" ]`.

PS: A while ago, I added a list of [default drivers](https://nixos.wiki/wiki/Mesa#Default_drivers) in the NixOS Wiki, that explains what "auto" covers. It's worth also mentioning (for future readers) that `zink`, `virgl` and `virtio-experimental` are useful but disabled by default in all scenarios.

Reply:
Did something a little bit fancier.

Reply:
This actually doesn't work because [the whole list is replaced](https://gitlab.freedesktop.org/mesa/mesa/-/blob/ef168a57b9c87cc3c120e5a750adde582dc2cad1/meson.build#L258-277) if `auto` exists in `vulkan_drivers`.